lib/HttpClient/CurlClient.php

```diff
@@ -439,4 +439,22 @@ private function canSafelyUseHttp2()
         $curlVersion = curl_version()['version'];
         return (version_compare($curlVersion, '7.60.0') >= 0);
     }
+
+    /**
+     * Checks if a list of headers contains a specific header name.
+     *
+     * @param string[] $headers
+     * @param string $name
+     * @return boolean
+     */
+    private function hasHeader($headers, $name)
+    {
+        foreach ($headers as $header) {
+            if (strncasecmp($header, "{$name}: ", strlen($name) + 2) === 0) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

Comment:
minor but isn't this quite slow/overhead? Should we maybe store "this information" (the fact that we set an idempotency key) elsewhere?

Reply:
Not really possible without modifying `ClientInterface` unfortunately.